### PR TITLE
[TrilinosApplication] Adding `CreateFastestAvailableDirectLinearSolver` to `trilinos_linear_solver_factory`, plus documentation

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR refactors the `trilinos_linear_solver_factory.py` module in the `TrilinosApplication`. The changes include adding documentation and adding `CreateFastestAvailableDirectLinearSolver`.

1. Added documentation comments using docstrings for various functions in the module to describe their purpose and usage.

2. Introduced a function `CreateFastestAvailableDirectLinearSolver()` to create and return the fastest available direct linear solver based on predefined order and solver availability. 

3. Update checks related with Amesos2 solvers

**🆕 Changelog**

- [Adding `CreateFastestAvailableDirectLinearSolver` to `trilinos_linear_solver_factory`, plus documentation](https://github.com/KratosMultiphysics/Kratos/commit/752d433e159f0079fb396f1fe0c0dd3bcfc3ca64)

### CODE REVIEW

File: applications/TrilinosApplication/python_scripts/trilinos_linear_solver_factory.py [hunk 1]
```diff
@@ -29,18 +59,131 @@ def _CheckIfTypeIsDeprecated(config):
...
+    Returns:
+        ConstructSolver: The fastest available direct linear solver.
+
+    Raises:
+        Exception: If no linear solver could be constructed.
+
+    Example:
+        fast_solver = CreateFastestAvailableDirectLinearSolver()
+    """
+    linear_solvers_by_speed = [
```
Comment (philbucher) [inline applications/TrilinosApplication/python_scripts/trilinos_linear_solver_factory.py:153]: Are you sure about the order?

I think you are right, but did you test it?

**Comment by matekelemen** [comment]
I'm not an expert on linear solvers, but as far as I know, each linear solver shines for a specific type of problem, and their accuracy/performance/robustness vary greatly for different systems. Isn't implying that a solver is "fastest" an oversimplification?

**Comment by philbucher** [comment]
> I'm not an expert on linear solvers, but as far as I know, each linear solver shines for a specific type of problem, and their accuracy/performance/robustness vary greatly for different systems. Isn't implying that a solver is "fastest", without knowing anything about the target problem, an oversimplification?

with iterative solvers sure, but with direct ones we had good experiences in serial with the same approach. Sometimes (well actually almost always) you just want any linear solver, and since this can have a huge impact on the solution times, it is IMO a good workflow to make an educated guess

File: applications/TrilinosApplication/python_scripts/trilinos_linear_solver_factory.py (hunk 1 above)
Comment (loumalouomega) [reply applications/TrilinosApplication/python_scripts/trilinos_linear_solver_factory.py:153]: I test it in my machine, but accordign to docs depends on several things, like system size and the graph sparsity. I copied/pasted official doc for documentation

**Comment by loumalouomega** [comment]
> I'm not an expert on linear solvers, but as far as I know, each linear solver shines for a specific type of problem, and their accuracy/performance/robustness vary greatly for different systems. Isn't implying that a solver is "fastest", without knowing anything about the target problem, an oversimplification?

I copied/pasted the official documentation. My doubth is SuperLUDist/MUMPS, in my tests MUMPS is faster, but accordign to documentation in depends of the problem

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
Finally 

imagen

**Pull request merged**